### PR TITLE
Make the config.js extendable

### PR DESCRIPTION
**Pull request by timlevett**
... by injecting things in my-app/app-config.js

This way you can add in a paths or shims into your application, but not rewrite the whole thing. See the changes to /docs for an example.

### CODE REVIEW

File: uw-frame-components/config.js [hunk 1]
```diff
@@ -1,51 +1,71 @@
 'use strict';
-define([], function() {
-  return {
+define(['./my-app/app-config'], function(myAppConfig) {
```
Comment (paulerickson) [inline uw-frame-components/config.js]: For reasons I don't fully grok, I think you need to add the ".js" suffix in situations like this where the runtime is not fully bootstrapped.

Comment (timlevett) [inline uw-frame-components/config.js]: Nice catch @paulerickson! Fixed, which also fixed the test run.

**Comment by vertein** [comment]
:+1:

**Comment by timlevett** [comment]
See http://uw-madison-doit.github.io/uw-frame/v2.6.0/#/md/injecting-dependencies for details

**Pull request merged**